subsurface-mobile-main.cpp

```diff
@@ -19,14 +19,48 @@
 #include <QLocale>
 #include <git2.h>
 
+// Implementation of STP logging 
+#include "core/ssrf.h"
+#ifdef ENABLE_STARTUP_TIMING
+#include <QElapsedTimer>
+void log_stp(const char *ident, void *buf)
+{
+static bool firstCall = true;
```

Comment:
The compiler has to do some kind of locking under the hood, because it has to check if this is the first invocation of the function and then construct the objects.

Of course the performance impacts are completely trivial, but I think these objects are fundamentally not function-local.